src/py/_pyodide/_importhook.py

```diff
@@ -131,3 +131,41 @@ def register_js_finder() -> None:
     global JsProxy
     JsProxy = _pyodide_core.JsProxy
     sys.meta_path.append(jsfinder)
+
+
+class StdlibFinder(MetaPathFinder):
+    def __init__(self) -> None:
+        # `test`` is not a stdlib module, but we unvendors in anyway.
+        self.stdlibs = sys.stdlib_module_names | {"test"}
+
+        # TODO: put list of unvendored stdlibs to somewhere else?
+        self.unvendored_stdlibs = {"distutils", "test", "_ssl", "_lzma"} & self.stdlibs
+
+    def find_spec(
+        self,
+        fullname: str,
+        path: Sequence[bytes | str] | None,
+        target: ModuleType | None = None,
+    ) -> ModuleSpec | None:
+        [parent, _, _] = fullname.partition(".")
+
+        if not parent or parent in sys.modules or parent not in self.stdlibs:
```

Comment:
And so if it's in sys.modules another loader should handle it? Maybe worth a comment that generally when this branch is for continuing to default import resolution.

Reply:
> And so if it's in sys.modules another loader should handle it

That condition is needed when someone makes a mistake such as:

```
import os.submodule_that_not_exists
```

The default importer will fail (return None) and it will fall back to this importer.